core/src/View/Kokkos_ViewTraits.hpp

```diff
@@ -168,8 +168,46 @@ struct ViewUniformType;
 }
 }  // namespace Kokkos
 
+// =========================================================
+// View template argument parsing
+// =========================================================
 namespace Kokkos {
 
+// =========================================================
+// Customization points for View for projects such as Sacado
+// =========================================================
+
+namespace Impl {
+// Type list for View Arguments
+template <class ValueType, class ArrayLayout, class DeviceType,
+          class MemoryTraits>
+struct ViewArguments {
+  using value_type    = ValueType;
+  using array_layout  = ArrayLayout;
+  using device_type   = DeviceType;
+  using memory_traits = MemoryTraits;
+  static_assert(!std::is_pointer_v<ValueType>);
+  static_assert(is_array_layout_v<ArrayLayout>);
+  static_assert(is_device_v<DeviceType>);
+  static_assert(is_memory_traits_v<MemoryTraits>);
+};
+
+// Customized mdspan/BasicView arguments
+template <class IndexType, class AccessorType>
+struct ViewCustomArguments {
+  using index_type    = IndexType;
+  using accessor_type = AccessorType;
+  static_assert(std::is_integral_v<IndexType>);
+};
```

Comment:
I still think these should be public and not in `Impl`, but potentially in `Experimental` or so.

Reply:
We may do so eventually - and likely need to. But I want to do that post Sacado working. And I don't want anyone else trying to really use this until then.